test/server/cards/02_SHD/leaders/HanSoloWorthTheRisk.spec.ts

```diff
@@ -0,0 +1,115 @@
+describe('Han Solo, Worth the Risk', function () {
+    integration(function (contextRef) {
+        describe('Han Solo\'s leader undeployed ability', function () {
+            it('should play a unit from our hand, it costs 1 resource less and take 2 damage', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['cantina-braggart', 'vanguard-infantry', 'green-squadron-awing', 'devotion'],
+                        groundArena: ['colonel-yularen#isb-director'],
+                        leader: 'han-solo#worth-the-risk',
+                        base: { card: 'echo-base', damage: 5 },
+                        resources: 4,
+                    },
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.hanSolo);
+                expect(context.player1).toBeAbleToSelectExactly([context.cantinaBraggart, context.vanguardInfantry, context.greenSquadronAwing]);
+                context.player1.clickCard(context.cantinaBraggart);
+
+                expect(context.player1.exhaustedResourceCount).toBe(0);
+                expect(context.cantinaBraggart).toBeInZone('groundArena');
+                expect(context.cantinaBraggart.damage).toBe(2);
+                expect(context.hanSolo.exhausted).toBeTrue();
+
+                context.hanSolo.exhausted = false;
+                context.player2.passAction();
+
+                context.player1.clickCard(context.hanSolo);
+                context.player1.clickCard(context.vanguardInfantry);
+
+                // choose which unit to give an experience token (vanguard infantry when defeated ability)
+                expect(context.player1).toBeAbleToSelectExactly([context.cantinaBraggart, context.colonelYularen]);
```

Comment:
I'm pretty sure this is incorrect due to an open issue with PlayCardSystem. If both triggered abilities from Vanguard Infantry and Yularen go into the same window, there's supposed to be a prompt asking which one to do first (see TriggeredAbilityWindow.spec.ts). If they're happening automatically in a certain order, that means they're not going into the same trigger window like they're supposed to.

Probably what's happening is that the play card ability is resolving as if it were nested, so it's resolving its triggers immediately instead of at the end of the card ability. Ennis I think is planning to correctly add that feature sometime soon, probably we should ask them in the discord what their plan is and go from there. We probably can leave this as a TODO for Ennis to finish.